backend/src/election/mod.rs

```diff
@@ -103,17 +106,156 @@ pub async fn election_status(
     Ok(Json(ElectionStatusResponse { statuses }))
 }
 
+struct ResultsInput {
+    election: Election,
+    polling_stations: Vec<PollingStation>,
+    results: Vec<(PollingStation, PollingStationResults)>,
+    summary: ElectionSummary,
+    creation_date_time: chrono::DateTime<chrono::Local>,
+}
+
+impl ResultsInput {
+    async fn new(
+        election_id: u32,
+        elections_repo: Elections,
+        polling_stations_repo: PollingStations,
+        polling_station_results_entries_repo: PollingStationResultsEntries,
+    ) -> Result<ResultsInput, APIError> {
+        let election = elections_repo.get(election_id).await?;
+        let polling_stations = polling_stations_repo.list(election.id).await?;
+        let results = polling_station_results_entries_repo
+            .list_with_polling_stations(polling_stations_repo, election.id)
+            .await?;
+
+        Ok(ResultsInput {
+            summary: ElectionSummary::from_results(&election, &results)?,
+            creation_date_time: chrono::Local::now(),
+            election,
+            polling_stations,
+            results,
+        })
+    }
+
+    fn as_xml(&self) -> EML510 {
+        EML510::from_results(
+            &self.election,
+            &self.results,
+            &self.summary,
+            &self.creation_date_time,
+        )
+    }
+
+    fn xml_filename(&self) -> String {
+        use chrono::Datelike;
+        format!(
+            "Telling_{}{}_{}.xml",
+            self.election.category.to_eml_code(),
+            self.election.election_date.year(),
+            self.election.location.replace(" ", "_"),
+        )
+    }
+
+    fn pdf_filename(&self) -> String {
+        use chrono::Datelike;
+        format!(
+            "PV_{}{}_{}.pdf",
+            self.election.category.to_eml_code(),
+            self.election.election_date.year(),
+            self.election.location.replace(" ", "_"),
+        )
+    }
+
+    fn zip_filename(&self) -> String {
+        use chrono::Datelike;
+        format!(
+            "election_result_{}{}_{}.zip",
+            self.election.category.to_eml_code(),
+            self.election.election_date.year(),
+            self.election.location.replace(" ", "_"),
+        )
+    }
+
+    fn into_pdf_model(self, xml_hash: impl Into<String>) -> PdfModel {
+        PdfModel::ModelNa31_2(ModelNa31_2Input {
+            polling_stations: self.polling_stations,
+            summary: self.summary,
+            election: self.election,
+            hash: xml_hash.into(),
+            creation_date_time: self.creation_date_time.format("%d-%m-%Y %H:%M").to_string(),
+        })
+    }
+}
+
+/// Download a zip containing a PDF for the PV and the EML with election results
+#[utoipa::path(
+    get,
+    path = "/api/elections/{election_id}/download_zip_results",
+    responses(
+        (
+            status = 200,
+            description = "ZIP",
+            content_type = "application/x-zip",
```

Comment:
The [official MIME type](https://www.iana.org/assignments/media-types/media-types.xhtml) is `application/zip`